framework/include/utils/StreamArguments.h

```diff
@@ -12,17 +12,35 @@
 /***
  * Streams all of the given arguments into the given stream
  */
-
 template <class StreamType>
 void
 streamArguments(StreamType &)
 {
 }
 
+// Recursive parameter pack expansion
 template <class StreamType, typename T, typename... Args>
 void
 streamArguments(StreamType & ss, T && val, Args &&... args)
 {
   ss << val;
   streamArguments(ss, std::forward<Args>(args)...);
 }
+
+// Base case for Tuple expansion recursive method
+template <std::size_t I = 0, class StreamType, typename... Args>
```

Comment:
It's funny that [the standard](https://en.cppreference.com/w/cpp/language/template_parameters#Default_template_arguments) specifies that non-default template parameters can follow defaults for function templates but not for class templates